projects/batfish/src/test/java/org/batfish/bdp/AbstractRibTest.java

```diff
@@ -0,0 +1,154 @@
+  private StaticRoute _mostGeneralRoute;
+
+  @Before
+  public void setupEmptyRib() {
+    _rib = new StaticRib(null);
+    _mostGeneralRoute = new StaticRoute(new Prefix("0.0.0.0/0"), Ip.ZERO, null, 0, 0);
+  }
+
+  @Test
+  public void testRibConstructor() {
+    // Test: the setupEmptyRib() fixture
+
+    // Assertions: Ensure that a new rib is empty upon construction
+    assertThat(_rib.getRoutes(), is(emptyIterableOf(StaticRoute.class)));
+    assertThat(_rib.containsRoute(_mostGeneralRoute), is(false));
+  }
+
+  @Test
+  public void testSingleRouteAdd() {
+    // Test: Adding one route
+    _rib.mergeRoute(_mostGeneralRoute);
+
+    // Assertions
+    // Check routes size
+    assertThat(_rib.getRoutes().size(), is(1));
+    // Check that containsRoute works as expected for this simple case
+    assertThat(_rib.containsRoute(_mostGeneralRoute), is(true));
+    assertThat(
+        _rib.containsRoute(new StaticRoute(new Prefix("1.1.1.1/32"), Ip.ZERO, null, 0, 0)),
+        is(false));
+  }
+
+  /** Inserts overlapping routes into the RIB and returns a (manually) ordered list of them */
+  private List<StaticRoute> setupOverlappingRoutes() {
+    // Setup helper
+    String[] testPrefixes =
+        new String[] {"10.0.0.0/8", "10.0.0.0/9", "10.128.0.0/9", "10.1.1.1/32"};
+    List<StaticRoute> routes = new ArrayList<>();
+
+    // Test: merge the routes into the RIB
+    for (String prefixStr : testPrefixes) {
+      StaticRoute r = new StaticRoute(new Prefix(prefixStr), Ip.ZERO, null, 0, 0);
+      _rib.mergeRoute(r);
+      routes.add(r);
+    }
+    return routes;
+  }
+
+  @Test
+  public void testRepeatedAdd() {
+    // Setup
+    StaticRoute route = new StaticRoute(new Prefix("10.0.0.0/11"), Ip.ZERO, null, 0, 0);
+
+    // Test: add the same route multiple times
+    for (int i = 0; i < 5; i++) {
+      _rib.mergeRoute(route);
+    }
+
+    // Assertions
+    // Ensure only one route is stored
+    assertThat(_rib.getRoutes().size(), is(1));
+    // Check that containsRoute works as expected for this simple case
+    assertThat(_rib.containsRoute(route), is(true));
+  }
+
+  @Test
+  public void testNonOverlappingRouteAdd() {
+    // Setup
+    StaticRoute r1 = new StaticRoute(new Prefix("1.1.1.1/32"), Ip.ZERO, null, 0, 0);
+    StaticRoute r2 = new StaticRoute(new Prefix("128.1.1.1/32"), Ip.ZERO, null, 0, 0);
+
+    // Test:
+    _rib.mergeRoute(r1);
+    _rib.mergeRoute(r2);
+
+    // Assertions
+    // Check that both routes exist
+    Set<StaticRoute> collectedRoutes = _rib.getRoutes();
+    assertThat(collectedRoutes.size(), is(2));
+    assertThat(_rib.containsRoute(r1), is(true));
+    assertThat(_rib.containsRoute(r2), is(true));
+    // Also check route collection via getRoutes()
+    assertThat(collectedRoutes, hasItem(r1));
+    assertThat(collectedRoutes, hasItem(r2));
+  }
+
+  @Test
+  public void testMultiOverlappingRouteAdd() {
+    // Setup/Test: Add multiple routes with overlapping prefixes
```

Comment:
This type of comment where useful for understanding test goals should go in the function javadoc.